aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -58,9 +58,10 @@ static void apply_batched_inverse_lib(Tensor& self, Tensor& self_inv, Tensor& in
       can_start.record(main_stream);
       can_start.block(main_stream);
 
-      int* pivot = reinterpret_cast<int*>(allocator.allocate(sizeof(int) * n).get());
+      auto dataPtr = allocator.allocate(sizeof(int) * n);
       _apply_single_inverse_helper<scalar_t>(
-        &self_data[i * self_mat_stride], &self_inv_data[i * self_inv_mat_stride], pivot, p_infos + i, n);
+        &self_data[i * self_mat_stride], &self_inv_data[i * self_inv_mat_stride],
+        reinterpret_cast<int*>(dataPtr.get()), p_infos + i, n);
```

Comment:
this predates this pr, but, shouldn't need a reinterpret cast here, static cast from void* should be fine